include/boost/geometry/algorithms/detail/disjoint/segment_box.hpp

```diff
@@ -181,12 +181,31 @@ struct disjoint_segment_box_sphere_or_spheroid
         azimuth_strategy.apply(lon1, lat1, b_lon_min, b_lat_max, a_b2);
         azimuth_strategy.apply(lon1, lat1, b_lon_max, b_lat_max, a_b3);
 
-        bool b0 = formula::azimuth_side_value(alp1, a_b0) > 0;
-        bool b1 = formula::azimuth_side_value(alp1, a_b1) > 0;
-        bool b2 = formula::azimuth_side_value(alp1, a_b2) > 0;
-        bool b3 = formula::azimuth_side_value(alp1, a_b3) > 0;
+        auto b0 = formula::azimuth_side_value(alp1, a_b0);
+        auto b1 = formula::azimuth_side_value(alp1, a_b1);
+        auto b2 = formula::azimuth_side_value(alp1, a_b2);
+        auto b3 = formula::azimuth_side_value(alp1, a_b3);
```

Comment:
The result type is known native type - `int`. It doesn't required to instantiate any template. It's even shorter than `auto`. I think that in such cases `auto` shouldn't be used.

Reply:
Btw, since you moved from `bool` to `int` these variables are now side value so maybe it'd be good to rename them to `s0`, `s1`, etc.?